src/Components/Components/src/EventCallbackFactoryBinderExtensions.cs

```diff
@@ -642,5 +697,139 @@ private static DateTime ConvertDateTime(object obj, string format)
             };
             return factory.Create<UIChangeEventArgs>(receiver, callback);
         }
+
+        // We can't rely on generics + static to cache here unfortunately. That would require us to overload
+        // CreateBinder on T : struct AND T : class, which is not allowed.
+        private static class BinderConverterCache
+        {
+            private readonly static ConcurrentDictionary<Type, Delegate> _cache = new ConcurrentDictionary<Type, Delegate>();
+
+            private static MethodInfo _convertToEnum;
+            private static MethodInfo _convertToNullableEnum;
+
+            public static BindConverter<T> Get<T>()
+            {
+                if (!_cache.TryGetValue(typeof(T), out var converter))
+                {
+                    // We need to replicate all of the primitive cases that we handle here so that they will behave the same way.
+                    // The result will be cached.
```

Comment:
I'm not super happy about this detail, but I think we mostly have covered the set of things we want to special case already.

Reply:
We only really want to special case something if we need to behave in a special way:
- supporting a format string, which type converters don't support
- supporting something without a type converter (nothing comes to mind, we could do Guid if we were really motivated)
- avoiding boxing (maybe DateTimeOfffset?)